src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -305,140 +308,142 @@ public void InitializeService(ServiceHost serviceHost)
-                    ScalingFactor = parameters.ScalingFactor / 100.0,
-                    TargetEnvironment = TargetEnvironmentType.Production
-                };
-    
-                // generate SQL DB recommendations, if applicable
-                if (parameters.TargetPlatforms.Contains("AzureSqlDatabase"))
-                {
-                    Stopwatch sqlDbStopwatch = new Stopwatch();
-                    sqlDbStopwatch.Start();
-
-                    prefs.EligibleSkuCategories = GetEligibleSkuCategories("AzureSqlDatabase", parameters.IncludePreviewSkus);
-                    resultSet.sqlDbResults = provider.GetSkuRecommendation(prefs, req);
+        internal RecommendationResultSet GenerateBaselineRecommendations(SqlInstanceRequirements req, GetSkuRecommendationsParams parameters)
+        {
+            RecommendationResultSet resultSet = new RecommendationResultSet();
 
-                    if (resultSet.sqlDbResults.Count < parameters.DatabaseAllowList.Count)
-                    {
-                        // if there are fewer recommendations than expected, find which databases didn't have a result generated and create a result with a null SKU
-                        List<string> databasesWithRecommendation = resultSet.sqlDbResults.Select(db => db.DatabaseName).ToList();
-                        foreach (var databaseWithoutRecommendation in parameters.DatabaseAllowList.Where(db => !databasesWithRecommendation.Contains(db)))
-                        {
-                            resultSet.sqlDbResults.Add(new SkuRecommendationResult()
-                            {
-                                //SqlInstanceName = sqlDbResults.FirstOrDefault().SqlInstanceName,
-                                SqlInstanceName = parameters.TargetSqlInstance,
-                                DatabaseName = databaseWithoutRecommendation,
-                                TargetSku = null,
-                                MonthlyCost = null,
-                                Ranking = -1,
-                                PositiveJustifications = null,
-                                NegativeJustifications = null,
-                            });
-                        }
-                    }
+            SkuRecommendationServiceProvider provider = new SkuRecommendationServiceProvider(new AzureSqlSkuBillingServiceProvider());
+            AzurePreferences prefs = new AzurePreferences()
+            {
+                EligibleSkuCategories = null,       // eligible SKU list will be adjusted with each recommendation type
+                ScalingFactor = parameters.ScalingFactor / 100.0,
+                TargetEnvironment = TargetEnvironmentType.Production
+            };
 
-                    sqlDbStopwatch.Stop();
-                    resultSet.sqlDbDurationInMs = sqlDbStopwatch.ElapsedMilliseconds;
+            // generate SQL DB recommendations, if applicable
+            if (parameters.TargetPlatforms.Contains("AzureSqlDatabase"))
```

Comment:
is there an enum / constant for "AzureSqlDatabase" to use instead.

Reply:
Agree, I didn't modify this code, only changed the line for the opening `{` while applying format to the document. We may need to create an item to remove all hardcoded strings in this file.